motoman_driver/include/motoman_driver/industrial_robot_client/motoman_utils.h

```diff
@@ -34,25 +34,25 @@
 
 #include <sstream>
 #include <iostream>
+#include <string>
+#include <map>
+
+#include "motoman_driver/industrial_robot_client/robot_group.h"
 
 namespace industrial_robot_client
 {
 namespace motoman_utils
 {
-
 /**
-  @brief Common error message displayed when node fails to find 'topic_list' parameter
-  */
+ * \brief Parses multi-group joints from topic_param
+ *
+ * \param topic_param base parameter name that holds the joint group listing
+ * \param robot_groups returns map of robot groups parsed from topic parameter
+ *
+ * \return true on success, false if parameter not found
+ */
+bool getJointGroups(std::string topic_param, std::map<int, RobotGroup> robot_groups);
```

Comment:
I think `robot_groups` should be a reference, as it's now passed by value. This causes callers to never get their `robot_groups` variables updated by this method, resulting in zero groups everywhere.